node/external/blockAPI/internalBlock.go

```diff
@@ -168,6 +170,102 @@ func (ibp *internalBlockProcessor) GetInternalStartOfEpochMetaBlock(format commo
 	return ibp.convertMetaBlockBytesByOutputFormat(format, blockBytes)
 }
 
+// GetInternalStartOfEpochValidatorsInfo wil return the epoch start validators info for the provided epoch
+func (ibp *internalBlockProcessor) GetInternalStartOfEpochValidatorsInfo(epoch uint32) ([]*state.ShardValidatorInfo, error) {
+	if ibp.selfShardID != core.MetachainShardId {
+		return nil, ErrMetachainOnlyEndpoint
+	}
+
+	storer, err := ibp.store.GetStorer(dataRetriever.MetaBlockUnit)
+	if err != nil {
+		return nil, err
+	}
+
+	epochStartIdentifier := core.EpochStartIdentifier(epoch)
+	blockBytes, err := storer.GetFromEpoch([]byte(epochStartIdentifier), epoch)
+	if err != nil {
+		return nil, err
+	}
+
+	metaBlock := &block.MetaBlock{}
+	err = ibp.marshalizer.Unmarshal(metaBlock, blockBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	validatorsInfo, err := ibp.getAllValidatorsInfo(metaBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	return validatorsInfo, nil
+}
+
+func (ibp *internalBlockProcessor) getAllValidatorsInfo(metaBlock data.HeaderHandler) ([]*state.ShardValidatorInfo, error) {
+	allValidatorInfo := make([]*state.ShardValidatorInfo, 0)
```

Comment:
will this slice include metachain validators, as well?

Reply:
yes, it will, it takes the validators info based on peer miniblocks info